docs/guides/advanced/authentication.md

```diff
@@ -4,7 +4,523 @@ title: Authentication
+    let endpoint = context.endpoint;
+    let url = endpoint + "/api/v1/users/me";
+    // ...
+  },
+});
+```
+
+Alternatively you can use a relative URL, and it will have the endpoint URL automatically prepended.
+
+```ts
+pack.addFormula({
+  // ...
+  execute: async function ([], context) {
+    // The endpoint URL will be automatically prepended.
+    let url = "/api/v1/users/me";
+    // ...
+  },
+});
+```
+
+
+### Entering manually
+
+To require the user to enter the endpoint URL, set the `requiresEndpointUrl` field to `true`. For additional validation you can also set the `endpointDomain` field to the root domain that all subdomains share,
+
+```ts
+pack.setUserAuthentication({
+  // ...
+  requiresEndpointUrl: true,
+  endpointDomain: "example.com",
+});
+```
+
+!!! warning "Endpoint domain must match network domain"
+    The `endpointDomain` field is used to validate the endpoint URL entered by the user, but doesn't impact which domains your Pack can make requests to. This is controlled by the separate [network domain][network_domains], and in practice these two domains are often set to the same value.
+
+When the user connects to their account they will now also be asked to provide the endpoint URL for their account.
+
+<img src="../../../images/auth_endpoint.png" srcset="../../../images/auth_endpoint_2x.png 2x" class="screenshot" alt="Setting the endpoint URL">
+
+[View Sample Code][sample_manual_endpoint]{ .md-button }
+
+!!! warning "Not compatible with OAuth2"
+    Packs that use OAuth2 authentication don't support a manual prompt for the endpoint URL. They should instead use one of the alternate options listed below.
+
+
+### Extracting from token exchange
+
+Some APIs that use OAuth2 authentication return the endpoint URL during the token exchange, as a custom property within the JSON body:
+
+```json
+{
+  "access_token": "...",
+  "token_type": "bearer",
+  "site_url": "https://vandelay.example.com"
+}
+```
+
+You can automatically extract this value and use it as the endpoint URL by setting the `endpointKey` field of the authentication configuration.
+
+```ts
+pack.setUserAuthentication({
+  // ...
+  endpointKey: "site_url",
+});
+```
+
+[View Sample Code][sample_automatic_endpoint]{ .md-button }
+
+
+### Prompting for user selection
+
+If the service allows the same user account to access multiple endpoints, and those endpoints can be determined with an API call, then you can prompt the user to select the endpoint they wish to connect to. This is accomplished using a `postSetup` step of the type `SetEndpoint`.
+
+```ts
+pack.setUserAuthentication({
+  // After approving access, the user should select which instance they want to
+  // connect to.
+  postSetup: [{
+    type: coda.PostSetupType.SetEndpoint,
+    name: "SelectEndpoint",
+    description: "Select the site to connect to:",
+    // Generate the list of endpoint options.
+    getOptionsFormula: coda.makeMetadataFormula(async function (context) {
```

Comment:
I don't think you should need `makeMetadataFormula` here or anywhere else anymore. Can you see if it works without it? If for some reason it needs it, we can fix it pretty easily.

Reply:
Ya, you do need it here. I have a TODO to file a bug about this.

Reply:
Filed: https://golinks.io/bug/20669

Reply:
Now that this was fixed (#1605) I've removed it from here and the Jira sample.